paddle/fluid/framework/details/inplace_op_pass.cc

```diff
@@ -78,6 +78,13 @@ const std::string kInplacedOpWhiteList[] = { // NOLINT
     "elementwise_add",
     "elementwise_add_grad",
 };
+
+// FIXME(zjl): Shapes of in-out of some ops are exactly the same,
+// but the static size during compiling time would be wrong.
+// Use a flag to indicate such ops. Please fix me when found a better way.
+static const std::unordered_set<std::string> kSameShapeOpWhiteSet{ // NOLINT
+    "reshape2"
+};
```

Comment:
Why? not understand what's the function of `kSameShapeOpWhiteSet`.

Reply:
Currently, inplace pass would only make in-out inplace when their compile-time shapes are the same. But in some nlp tasks, the compile-time shapes of in-out of reshape op are not the same. For example, shape of X is [-1, 3, 30] and shape of Out would be [-1, 30]. But in-out of some ops (such as reshape op) are exactly the same, regardless the mismatching compile-time shapes.

Reply:
I see, thanks.